pkg/certificate/providers/tresor/fake/fake.go

```diff
@@ -17,14 +18,63 @@ import (
 
 const (
 	rootCertOrganization = "Open Service Mesh Tresor"
+	initialRootName      = "osm-mesh-root-certificate"
 )
 
-type fakeMRCClient struct{}
+type fakeMRCClient struct {
+	mrcChannel chan certificate.MRCEvent
+}
+
+// NewFakeMRC allows for publishing events on to the watch channel to generate MRC events
+func NewFakeMRC() *fakeMRCClient {
+	ch := make(chan certificate.MRCEvent)
 
+	return &fakeMRCClient{
+		mrcChannel: ch,
+	}
+}
+
+// NewCertEvent allows pushing MRC events which can trigger cert changes
+func (c *fakeMRCClient) NewCertEvent(name, state string) {
+	c.mrcChannel <- certificate.MRCEvent{
+		Type: certificate.MRCEventAdded,
+		MRC: &v1alpha2.MeshRootCertificate{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: "osm-system",
+			},
+			Spec: v1alpha2.MeshRootCertificateSpec{
+				TrustDomain: "cluster.local",
+				Provider: v1alpha2.ProviderSpec{
+					Tresor: &v1alpha2.TresorProviderSpec{
+						CA: v1alpha2.TresorCASpec{
+							SecretRef: v1.SecretReference{
+								Name:      "osm-ca-bundle",
+								Namespace: "osm-system",
+							},
+						},
+					},
+				},
+			},
+			Status: v1alpha2.MeshRootCertificateStatus{
+				State: state,
+			},
+		},
+	}
+}
+
+// GetCertIssuerForMRC will return a root cert for testing. If called with MRCStateIssuingRollout it creates a
+// new cert with a different issuer name based on the MRC name.
 func (c *fakeMRCClient) GetCertIssuerForMRC(mrc *v1alpha2.MeshRootCertificate) (certificate.Issuer, pem.RootCertificate, error) {
 	rootCertCountry := "US"
 	rootCertLocality := "CA"
-	ca, err := tresor.NewCA("Fake Tresor CN", 1*time.Hour, rootCertCountry, rootCertLocality, rootCertOrganization)
+	cn := certificate.CommonName("Fake Tresor CN")
+	if mrc.Status.State == constants.MRCStateIssuingRollout {
```

Comment:
Could you elaborate on why the issueingRollout state needs to be handled uniquely? Could the id for every issuer be the mrc name?

Reply:
i believe it's to indicate that the issuer trust domain has changed, thereby changing the CN of the CA.

Although it doesn't compare the CA so there's actually nothing to do here.. the actual manager will handle that logic properly

(If i'm understanding the reasoning here)

Reply:
yes, I didn't want to put too much logic in here since it's a test function and this was enough to cause the rotation

Reply:
spoke with @jaellio who helped me understand the rotation logic a bit better and found this check isn't needed to fire the rotation so removed it.